File: xadditus-android/app/src/main/java/com/linoagli/android/xadditus/activities/MainActivity.java
/**
 * xadditus-android Project.
 * com.linoagli.android.xadditus.activities
 *
 * @author Faye-Lino Agli, username: linoagli
 */
package com.linoagli.android.xadditus.activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import com.linoagli.android.xadditus.AppRuntime;
import com.linoagli.android.xadditus.base.BaseActivity;
import com.linoagli.android.xadditus.services.AutoConnectService;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        startService(new Intent(this, AutoConnectService.class));

        if (AppRuntime.connectionsManager.isServiceActive()) {
            startActivity(new Intent(this, InputInterfaceActivity.class));
        } else {
            startActivity(new Intent(this, DeviceScanActivity.class));
        }

        finish();
    }
}

File: xadditus-android/app/src/main/kotlin/com/linoagli/android/xadditus/activities/MainActivity.kt
/**
 * xadditus-android Project.
 * com.linoagli.android.xadditus.activities
 * @author Faye-Lino Agli, username: linoagli
 */
package com.linoagli.android.xadditus.activities

import android.content.Intent
import android.os.Bundle
import com.linoagli.android.xadditus.Runtime
import com.linoagli.android.xadditus.activities.input.InputInterfaceActivity
import com.linoagli.android.xadditus.base.BaseActivity
import com.linoagli.android.xadditus.services.AutoConnectService

class MainActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        startService(Intent(this@MainActivity, AutoConnectService::class.java))

        val activityClass = if (Runtime.connectionsManager.isServiceActive) InputInterfaceActivity::class.java else DeviceScanActivity::class.java
        startActivity(Intent(this@MainActivity, activityClass))

        finish()
    }
}